core/src/stark/air.rs

```diff
@@ -0,0 +1,423 @@
+use std::marker::PhantomData;
+
+use crate::air::MachineAir;
+pub use crate::air::SP1AirBuilder;
+use crate::memory::MemoryChipKind;
+use crate::runtime::ExecutionRecord;
+use p3_air::Air;
+use p3_air::BaseAir;
+use p3_field::PrimeField32;
+
+pub use riscv_chips::*;
+
+pub(crate) mod riscv_chips {
+    pub use crate::alu::AddChip;
+    pub use crate::alu::BitwiseChip;
+    pub use crate::alu::DivRemChip;
+    pub use crate::alu::LtChip;
+    pub use crate::alu::MulChip;
+    pub use crate::alu::ShiftLeft;
+    pub use crate::alu::ShiftRightChip;
+    pub use crate::alu::SubChip;
+    pub use crate::bytes::ByteChip;
+    pub use crate::cpu::CpuChip;
+    pub use crate::field::FieldLTUChip;
+    pub use crate::memory::MemoryGlobalChip;
+    pub use crate::program::ProgramChip;
+    pub use crate::syscall::precompiles::blake3::Blake3CompressInnerChip;
+    pub use crate::syscall::precompiles::edwards::EdAddAssignChip;
+    pub use crate::syscall::precompiles::edwards::EdDecompressChip;
+    pub use crate::syscall::precompiles::k256::K256DecompressChip;
+    pub use crate::syscall::precompiles::keccak256::KeccakPermuteChip;
+    pub use crate::syscall::precompiles::sha256::ShaCompressChip;
+    pub use crate::syscall::precompiles::sha256::ShaExtendChip;
+    pub use crate::syscall::precompiles::weierstrass::WeierstrassAddAssignChip;
+    pub use crate::syscall::precompiles::weierstrass::WeierstrassDoubleAssignChip;
+    pub use crate::utils::ec::edwards::ed25519::Ed25519Parameters;
+    pub use crate::utils::ec::edwards::EdwardsCurve;
+    pub use crate::utils::ec::weierstrass::secp256k1::Secp256k1Parameters;
+    pub use crate::utils::ec::weierstrass::SWCurve;
+}
+
+pub enum RiscvAir<F> {
+    /// An AIR that containts a preprocessed program table and a lookup for the instructions.
+    Program(ProgramChip),
+    /// An AIR for the RISC-V CPU. Each row represents a cpu cycle.
+    Cpu(CpuChip),
+    /// An AIR for the RISC-V Add instruction.
+    Add(AddChip),
+    /// An AIR for the RISC-V Sub instruction.
+    Sub(SubChip),
+    /// An AIR for RISC-V Bitwise instructions.
+    Bitwise(BitwiseChip),
+    /// An AIR for RISC-V Mul instruction.
+    Mul(MulChip),
+    /// An AIR for RISC-V Div and Rem instructions.
+    DivRem(DivRemChip),
+    /// An AIR for RISC-V Lt instruction.
+    Lt(LtChip),
+    /// An AIR for RISC-V SLL instruction.
+    ShiftLeft(ShiftLeft),
+    /// An AIR for RISC-V SRL and SRA instruction.
+    ShiftRight(ShiftRightChip),
+    /// A lookup table for byte operations.
+    ByteLookup(ByteChip),
+    /// An table for `less than` operation on field elements.
+    FieldLTU(FieldLTUChip),
+    /// A table for initializing the memory state.
+    MemoryInit(MemoryGlobalChip),
+    /// A table for finalizing the memory state.
+    MemoryFinal(MemoryGlobalChip),
+    /// A table for initializing the program memory.
+    ProgramMemory(MemoryGlobalChip),
+
+    ShaExtend(ShaExtendChip),
```

Comment:
nit: the spacing here is a bit weird, can we just have a comment per element in the enum?